### PR TITLE
mnist scenario

### PR DESCRIPTION
**Pull request by avssridhar**

### CODE REVIEW

File: scenarios/mnist/ci/Dockerfile.mnist_1 [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+RUN apt-get upgrade && apt-get update \
+    && apt-get install -y python3 python3-pip \
+    && apt-get install -y openjdk-8-jdk
+
+RUN pip3 install pyspark pandas
+
+ENV JAVA_HOME /usr/lib/jvm/java-8-openjdk-amd64/
+RUN export JAVA_HOME
+
+COPY ccr_depa_covid_poc_dp_data_prep_cowin.py ccr_depa_covid_poc_dp_data_prep_cowin.py
```
Comment (kapilvgit) [inline scenarios/mnist/ci/Dockerfile.mnist_1]: Again, drop all references to icmr etc in this sample.

File: scenarios/mnist/ci/push-containers.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+containers=("preprocess-icmr:latest" "preprocess-cowin:latest" "preprocess-index:latest" "ccr-model-save:latest")
```
Comment (kapilvgit) [inline scenarios/mnist/ci/push-containers.sh]: remove references to icrm, cowin and index

File: scenarios/mnist/config/query_config.json
Comment (kapilvgit) [inline scenarios/mnist/config/query_config.json:1]: This query config does not seem right for this example.

File: .DS_Store
Comment (kapilvgit) [inline .DS_Store:1]: Why is this here?

File: scenarios/mnist/README.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+Assuming you have cleartext access to all the de-identified datasets, you can train the model as follows. 
+
+```bash
+./train.sh
+```
+The script joins the datasets using a configuration defined in [query_config.json](./config/query_config.json) and trains the model using a configuration defined in [model_config.json](./config/model_config.json). If all goes well, you should see output similar to the following output, and the trained model will be saved under the folder `/tmp/output`. 
+
+```
+docker-train-1  | {'input_dataset_path': '/tmp/sandbox_icmr_cowin_index_without_key_identifiers.csv', 'saved_model_path': '/mnt/remote/model/model.onnx', 'saved_model_optimizer': '/mnt/remote/model/dpsgd_model_opimizer.pth', 'saved_weights_path': '', 'batch_size': 2, 'total_epochs': 5, 'max_grad_norm': 0.1, 'epsilon_threshold': 1.0, 'delta': 0.01, 'sample_size': 60000, 'target_variable': 'icmr_a_icmr_test_result', 'test_train_split': 0.2, 'metrics': ['accuracy', 'precision', 'recall']}
```
Comment (kapilvgit) [inline scenarios/mnist/README.md:105]: We should not have references to icmr etc in this sample.

File: scenarios/mnist/contract/contract.json [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  "datasets": [
+    {
+      "id" : "19517ba8-bab8-11ed-afa1-0242ac120002",
+      "name": "icmr",
+      "url" : "https://ccrcontainer.blob.core.windows.net/mnist_1/data.img",
+      "provider": "",
+      "key" : {
+        "type": "azure",
+        "properties": {
+          "kid": "MNIST_1_FilesystemEncryptionKey",
```
Comment (kapilvgit) [inline scenarios/mnist/contract/contract.json:18]: There is an extract _ here, compared to other key identifiers.

File: scenarios/mnist/contract/contract.json [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  "schemaVersion": "0.1",
+  "startTime": "2023-03-14T00:00:00.000Z",
+  "expiryTime": "2024-03-14T00:00:00.000Z",
+  "tdc" : "",
+  "tdps" : [],
+  "ccrp": "did:web:ccrprovider.github.io",
+  "datasets": [
+    {
+      "id" : "19517ba8-bab8-11ed-afa1-0242ac120002",
+      "name": "icmr",
```
Comment (kapilvgit) [inline scenarios/mnist/contract/contract.json]: Change name.

File: scenarios/mnist/data/.DS_Store
Comment (kapilvgit) [inline scenarios/mnist/data/.DS_Store:1]: This should not be checked in.

File: scenarios/mnist/data/1-create-storage-containers.sh [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    --name $AZURE_RESOURCE_GROUP
+
+az storage account create \
+    --resource-group $AZURE_RESOURCE_GROUP \
+    --name $AZURE_STORAGE_ACCOUNT_NAME
+
+az storage container create \
+    --resource-group $AZURE_RESOURCE_GROUP \
+    --account-name $AZURE_STORAGE_ACCOUNT_NAME \
+    --name $AZURE_ICMR_CONTAINER_NAME
```
Comment (kapilvgit) [inline scenarios/mnist/data/1-create-storage-containers.sh]: No ICMR, COWIN or INDEX here.

File: scenarios/mnist/data/3-import-keys.sh [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    export BEARER_TOKEN=$(az account get-access-token --resource https://vault.azure.net | jq -r .accessToken)
+    echo "Importing keys to AKV key vaults can be only of type RSA-HSM"
+    export AZURE_AKV_KEY_TYPE="RSA-HSM"
+elif [[ "$AZURE_KEYVAULT_ENDPOINT" == *".managedhsm.azure.net" ]]; then
+    export BEARER_TOKEN=$(az account get-access-token --resource https://managedhsm.azure.net | jq -r .accessToken)    
+    export AZURE_AKV_KEY_TYPE="oct-HSM"
+fi
+
+DATADIR=`pwd`
+import_key "ICMRFilesystemEncryptionKey" $DATADIR/icmrkey.bin
```
Comment (kapilvgit) [inline scenarios/mnist/data/3-import-keys.sh]: No ICMR, COWIN or INDEX here.

File: scenarios/mnist/data/5-upload-encrypted-data.sh [hunk 8]
```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+ACCOUNT_KEY=$(az storage account keys list --account-name $AZURE_STORAGE_ACCOUNT_NAME --only-show-errors | jq -r .[0].value)
+
+az storage blob upload \
+  --account-name $AZURE_STORAGE_ACCOUNT_NAME \
+  --container $AZURE_ICMR_CONTAINER_NAME \
+  --file icmr.img \
```
Comment (kapilvgit) [inline scenarios/mnist/data/5-upload-encrypted-data.sh]: Please change names.

File: scenarios/mnist/data/mnist_1/.DS_Store
Comment (kapilvgit) [inline scenarios/mnist/data/mnist_1/.DS_Store:1]: Should not be checked in.

File: scenarios/mnist/data/mnist_1/poc_data_mnist_1_data.csv
Comment (kapilvgit) [inline scenarios/mnist/data/mnist_1/poc_data_mnist_1_data.csv:1]: Why are we checking this data in as CSV?

File: scenarios/mnist/data/output/ccr_depa_trg_model_logger.json
Comment (kapilvgit) [inline scenarios/mnist/data/output/ccr_depa_trg_model_logger.json:1]: This file should not be checked in.

File: scenarios/mnist/deployment/aci/arm-template.json [hunk 9]
```diff
@@ -0,0 +1,192 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "name": {
+      "defaultValue": "depa-training-covid",
```
Comment (kapilvgit) [inline scenarios/mnist/deployment/aci/arm-template.json]: Change name

File: scenarios/mnist/deployment/aci/deploy.sh [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+export CCE_POLICY_HASH=$(go run $TOOLS_HOME/securitypolicydigest/main.go -p $CCE_POLICY)
+echo "Training container policy hash $CCE_POLICY_HASH"
+
+export CONTRACTS=$contract
+export MODEL_CONFIGURATION=`cat $modelConfiguration | base64 --wrap=0`
+export QUERY_CONFIGURATION=`cat $queryConfiguration | base64 --wrap=0`
+
+function generate_encrypted_filesystem_information() {
+  end=`date -u -d "60 minutes" '+%Y-%m-%dT%H:%MZ'`
+  ICMR_SAS_TOKEN=$(az storage blob generate-sas --account-name $AZURE_STORAGE_ACCOUNT_NAME --container-name $AZURE_ICMR_CONTAINER_NAME --permissions r --name data.img --expiry $end --only-show-errors)
```
Comment (kapilvgit) [inline scenarios/mnist/deployment/aci/deploy.sh]: Change names.

File: scenarios/mnist/deployment/docker/docker-compose-train.yml [hunk 11]
```diff
@@ -0,0 +1,12 @@
+services:
+  train:
+    image: ${CONTAINER_REGISTRY}/depa-training:latest
+    volumes:
+      - $ICMR_INPUT_PATH:/mnt/remote/icmr
```
Comment (kapilvgit) [inline scenarios/mnist/deployment/docker/docker-compose-train.yml:5]: Rename paths.

File: scenarios/mnist/src/ccr_dpsgd_model_saving_template_v2.py [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+
+train_dataset = CustomDataset(train_features, train_target)
+val_dataset = CustomDataset(val_features, val_target)
+
+batch_size = 32
+train_loader = DataLoader(train_dataset, batch_size=batch_size, shuffle=True)
+val_loader = DataLoader(val_dataset, batch_size=batch_size)
+
+# Step 4: Define a simple neural network model
+class SimpleModel(nn.Module):
```
Comment (kapilvgit) [inline scenarios/mnist/src/ccr_dpsgd_model_saving_template_v2.py:51]: Where are you getting the model from?

**Review by kapilvgit — COMMENTED** [review]
Can you please add a top level description for the PR? Also mention what tests you have run.

File: scenarios/.DS_Store
Comment (kapilvgit) [inline scenarios/.DS_Store:1]: Remove this file.

File: scenarios/mnist/.DS_Store
Comment (kapilvgit) [inline scenarios/mnist/.DS_Store:1]: Remove this file.

File: scenarios/mnist/ci/Dockerfile.mnist_2 [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+RUN apt-get upgrade && apt-get update \
+    && apt-get install -y python3 python3-pip \
+    && apt-get install -y openjdk-8-jdk
+
+RUN pip3 install pyspark pandas
+
+ENV JAVA_HOME /usr/lib/jvm/java-8-openjdk-amd64/
+RUN export JAVA_HOME
+
+COPY ccr_depa_covid_poc_dp_data_prep_index.py ccr_depa_covid_poc_dp_data_prep_index.py
```
Comment (kapilvgit) [inline scenarios/mnist/ci/Dockerfile.mnist_2:14]: This file needs to be renamed, remove index.

File: scenarios/mnist/ci/Dockerfile.mnist_3 [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+RUN apt-get upgrade && apt-get update \
+    && apt-get install -y python3 python3-pip \
+    && apt-get install -y openjdk-8-jdk
+
+RUN pip3 install pyspark pandas
+
+ENV JAVA_HOME /usr/lib/jvm/java-8-openjdk-amd64/
+RUN export JAVA_HOME
+
+COPY ccr_depa_covid_poc_dp_data_prep_icmr.py ccr_depa_covid_poc_dp_data_prep_icmr.py
```
Comment (kapilvgit) [inline scenarios/mnist/ci/Dockerfile.mnist_3:14]: This file needs to be renamed, remove icmr

File: scenarios/mnist/data/.DS_Store
Comment (kapilvgit) [inline scenarios/mnist/data/.DS_Store:1]: Remove this file.

File: scenarios/mnist/data/mnist_1/.DS_Store
Comment (kapilvgit) [inline scenarios/mnist/data/mnist_1/.DS_Store:1]: Remove this file.

File: scenarios/mnist/data/mnist_2/.DS_Store
Comment (kapilvgit) [inline scenarios/mnist/data/mnist_2/.DS_Store:1]: Remove this file.

File: scenarios/mnist/data/output/ccr_depa_trg_model_logger.json
Comment (kapilvgit) [inline scenarios/mnist/data/output/ccr_depa_trg_model_logger.json:1]: This should not be checked in

File: scenarios/mnist/deployment/docker/docker-compose-train.yml (hunk 11 above)
Comment (kapilvgit) [reply scenarios/mnist/deployment/docker/docker-compose-train.yml:5]: This does not appear to be resolved yet.

File: scenarios/mnist/deployment/docker/train.sh
Comment (kapilvgit) [inline scenarios/mnist/deployment/docker/train.sh:1]: Names need changing here.
